src/components/Game/Slots/Slots.js

```diff
@@ -7,44 +7,30 @@ import { WORD_LENGTH } from '../../../constants/constants';
 import { generateID } from '../../../helpers';
 
 
-const Slots = ({ lettersInSlots, clicked }) => {
-    const slotsNodes = [];
-    let slotIndex = 0;
-
-    while (slotIndex < WORD_LENGTH) {
-        const letter = lettersInSlots && lettersInSlots[slotIndex];
-        const letterIndex = slotIndex;
-
-        slotsNodes.push(
-            <li
-                key={generateID()}
-                className={classes.Slot}
-            >
-                {
-                    letter
-                        ? (
-                            <span
-                                className={classes.SlotLetter}
-                                onClick={() => clicked(letter.id, letterIndex)}
-                            >
-                                {letter.value}
-                            </span>
-                        )
-                        : null
-                }
-            </li>,
-        );
-        slotIndex += 1;
-    }
-
-    return (
-        <section className={classes.SlotsSection}>
-            <ul className={classes.Slots}>
-                {slotsNodes}
-            </ul>
-        </section>
-    );
-};
+const Slots = ({ lettersInSlots, clicked }) => (
+    <section className={classes.SlotsSection}>
+        <ul className={classes.Slots}>
+
+            {[...Array(WORD_LENGTH).keys()].map((i) => {
+                const letter = lettersInSlots[i] || null;
+                return (
+                    <li key={generateID()} className={classes.Slot}>
```

Comment:
maybe better `key={letter.id}`?

Reply:
Not sure generating unique key each time render runs is a good practice...

Reply:
of course, `generateID` is a leftover after the previous version. but still I cannot use `letter.id` as a key since `letter` doesn't exist as long as `lettersInSlots` is empty (it is at the beginning). and I can't use index as a key (lint doesn't let me). so?
moreover what concerns me is that in this loop we always have 7 `<li>`s and usually less then 7 `letters`. what in such a sutuation? is it ok as far as the key is concerned?

Reply:
1. why lint does not let you? what rule is not met?
2. I don't see reason for concern, some slots may be empty, we need them to display a black line indicating they exist, later on we will need them for drag and drop functionality (e.g. it will be possible to drag empty slot)

Reply:
1. react/no-array-index-key. it's a warning, not an error, but correct me if I'm wrong: it won't pass circleCI tests, right? plus would it be safe to use index as a keys when we plan on changing slots' order?
2. slots being empty is not my concern. my concern is: if we use `letter.id` (let's assume we can do it) as a key, empty slots won't have their keys, will they?

Reply:
1.
they won't
most likely not, I think you're right

2.
yes, you're right, no `letter.id` no fun

so maybe leave it as it is or disable lint rule and use index key (right now we don't sort/order/rearrange)